### ISSUE TITLE
Add support for .eq_crt files

### ISSUE THREAD

**Issue by zentol** [Synthlight/MHW-Editor#1]
I have some info for .eq_crt files, which define the required crafting (not _upgrade_) materials.

The files start with a 10 byte preamble, followed by 37 byte entries for each item.
The preambles consists of:
```
fixed 6-bytes sequence (01 10 09 18 79 00)
int32 number_of_entries
```
An entry consists of:
```
uint8	item_category (charm, chest, helmet,..., weapon_class)
uint16	item_id (can be directly cross-referenced with armor.am_dat, weapons are a bit tricky)
uint16	item_class (e.g., all baan armors have the same value here; sometimes 0)
int32	? (sometimes variant for armors)? / charm "class" (health, paralysis, ...); -1 for many
uint32	event/update reference (festivals, iceborne)?
unit32	? (is set for bushi layered armored, brigade layered armor head and a bunch of entries in charm.eq_crt)
unit8	item_rank (regular armor: 0=LR, 1=HR, 2=MR, layered armor: 0=LR/HR, 1=MR)
uint24	? (always zero)
uint16	id_material_1
uint8	count_material_1
uint16	id_material_2
uint8	count_material_2
uint16	id_material_3
uint8	count_material_3
uint16	id_material_4
uint8	count_material_4
uint8	?
uint8	?
uint8	?
uint8	?
```

**Comment by Synthlight** [comment]
Here's a few screenshots of the file assuming I got everything right.
Poke me if anything looks off.
![armorEqCrt](https://user-images.githubusercontent.com/1075166/73603903-fe0d2600-4556-11ea-833c-c4e32f376d54.png)
![charmEqCrt](https://user-images.githubusercontent.com/1075166/73603904-fe0d2600-4556-11ea-8be3-3c8cd6a75c3f.png)
![wepEqCrt](https://user-images.githubusercontent.com/1075166/73603905-fea5bc80-4556-11ea-8074-c0315e11ef31.png)

**Comment by zentol** [comment]
One thing I got wrong is that the supposed Event/Update ref seems to be more of an story unlock condition.

But I see no error on your part; having the UI made the above issue a lot easier to figure out 👍

**Comment by Synthlight** [comment]
> uint8	item_category (charm, chest, helmet,..., weapon_class)

What is the full list here? I don't have an enum for this.

**Comment by zentol** [comment]
The item_category is dependent on the file;
- for weapon.eq_crt it is 0-13 (0=Greatsword, 1=Longsword, don't know the rest but could find out)
- for armor.eq_crt it is 0-5 (0=head, 1=chest, 2=waist, 3=arms, 4=legs, 5=charm)
- for charm.eq_crt it is 0
- for ot.eq_crt it is 0-2 (0=weapon, 1=head, 2=body)

**Comment by Synthlight** [comment]
Done. I left the other as unknown for now.

**Comment by zentol** [comment]
Here are the full weapon categories (corrections in bold):
0=Greatsword
**1=Sword&Shield**
2=Dual Blades
**3=Longsword**
4=Hammer
5=Hunting Horn
6=Lance
7=Gunlance
8=Switch Axe
9=Charge Blade
10=Insect Glaive
11=Bow
12=Light Bowgun
13=Heavy Bowgun

**Comment by Synthlight** [comment]
Done and released in v1.6.
